src/sagemaker/workflow/_utils.py

```diff
@@ -364,72 +358,84 @@ def __init__(
 
         self._properties = Properties(path=f"Steps.{name}", shape_name="DescribeModelPackageOutput")
 
+        if not self.step_args:
```

Comment:
I think this warning should be put into the top-level function, `RegisterModel` instead of here, users will not call this function `_RegisterModelStep ` directly.

Reply:
We marked the `RegisterModel` as deprecated, so as long as user uses the `ModelStep`, we should never reach here, and if they keep using `RegisterModel`, they will get deprecation warning anyway.

Reply:
Makes sense. Removed this warning in the new commit